File: server-scala-play-server/app/model/RatingIndexResponse.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for RatingIndexResponse.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class RatingIndexResponse(
  ratingIndexId: Option[Long],
  ratingCount: Option[Long],
  ratingAverage: Option[Float],
  value: Option[Float],
  display: Option[String],
  ratableType: Option[RatingIndexResponse.RatableType.Value],
  ratableId: Option[Long],
  category: Option[CategoryResponse],
  ratable: Option[JsObject],
  secondaryType: Option[String]
)

object RatingIndexResponse {
  implicit lazy val ratingIndexResponseJsonFormat: Format[RatingIndexResponse] = Json.format[RatingIndexResponse]

  // noinspection TypeAnnotation
  object RatableType extends Enumeration {
    val LOCATABLE = Value("LOCATABLE")
    val RESERVABLE = Value("RESERVABLE")
    val PERMISSIONABLE = Value("PERMISSIONABLE")
    val NOTABLE = Value("NOTABLE")
    val ASSETABLE = Value("ASSETABLE")
    val LIKABLE = Value("LIKABLE")
    val FLAGABLE = Value("FLAGABLE")
    val FAVORITABLE = Value("FAVORITABLE")
    val RATABLE = Value("RATABLE")
    val ALBUM = Value("ALBUM")
    val COLLECTION = Value("COLLECTION")
    val APPLICATION = Value("APPLICATION")
    val APPLICATIONSETTING = Value("APPLICATION_SETTING")
    val APPLICATIONCERT = Value("APPLICATION_CERT")
    val APPLICATIONPLACEMENT = Value("APPLICATION_PLACEMENT")
    val ACCOUNT = Value("ACCOUNT")
    val ACCOUNTSETTING = Value("ACCOUNT_SETTING")
    val GAMELEVEL = Value("GAME_LEVEL")
    val PACK = Value("PACK")
    val MISSION = Value("MISSION")
    val TOURNAMENT = Value("TOURNAMENT")
    val ASSET = Value("ASSET")
    val ALBUMCONTEST = Value("ALBUM_CONTEST")
    val THEMEDESCRIPTOR = Value("THEME_DESCRIPTOR")
    val OFFER = Value("OFFER")
    val OFFERLOCATION = Value("OFFER_LOCATION")
    val EVENT = Value("EVENT")
    val RETAILER = Value("RETAILER")
    val RETAILERLOCATION = Value("RETAILER_LOCATION")
    val NOTE = Value("NOTE")
    val CREATIVE = Value("CREATIVE")
    val FAVORITE = Value("FAVORITE")
    val LIKE = Value("LIKE")
    val RATING = Value("RATING")
    val ANALYTIC = Value("ANALYTIC")
    val THIRDPARTYCREDENTIAL = Value("THIRD_PARTY_CREDENTIAL")
    val THIRDPARTYNETWORK = Value("THIRD_PARTY_NETWORK")
    val REGION = Value("REGION")

    type RatableType = Value
    implicit lazy val RatableTypeJsonFormat: Format[Value] = Format(Reads.enumNameReads(this), Writes.enumNameWrites[this.type])
  }
}

File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/RatingIndexResponse.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class RatingIndexResponse 
{
    public long RatingIndexId { get; set; }
    public long RatingCount { get; set; }
    public float RatingAverage { get; set; }
    public float Value { get; set; }
    public string Display { get; set; }
    
    /// <summary>
    /// Gets or Sets RatableType
    /// </summary>
    public enum RatableTypeEnum
    {
        
        /// <summary>
        /// Enum LOCATABLEEnum for LOCATABLE
        /// </summary>
        LOCATABLEEnum = 1,
        
        /// <summary>
        /// Enum RESERVABLEEnum for RESERVABLE
        /// </summary>
        RESERVABLEEnum = 2,
        
        /// <summary>
        /// Enum PERMISSIONABLEEnum for PERMISSIONABLE
        /// </summary>
        PERMISSIONABLEEnum = 3,
        
        /// <summary>
        /// Enum NOTABLEEnum for NOTABLE
        /// </summary>
        NOTABLEEnum = 4,
        
        /// <summary>
        /// Enum ASSETABLEEnum for ASSETABLE
        /// </summary>
        ASSETABLEEnum = 5,
        
        /// <summary>
        /// Enum LIKABLEEnum for LIKABLE
        /// </summary>
        LIKABLEEnum = 6,
        
        /// <summary>
        /// Enum FLAGABLEEnum for FLAGABLE
        /// </summary>
        FLAGABLEEnum = 7,
        
        /// <summary>
        /// Enum FAVORITABLEEnum for FAVORITABLE
        /// </summary>
        FAVORITABLEEnum = 8,
        
        /// <summary>
        /// Enum RATABLEEnum for RATABLE
        /// </summary>
        RATABLEEnum = 9,
        
        /// <summary>
        /// Enum ALBUMEnum for ALBUM
        /// </summary>
        ALBUMEnum = 10,
        
        /// <summary>
        /// Enum COLLECTIONEnum for COLLECTION
        /// </summary>
        COLLECTIONEnum = 11,
        
        /// <summary>
        /// Enum APPLICATIONEnum for APPLICATION
        /// </summary>
        APPLICATIONEnum = 12,
        
        /// <summary>
        /// Enum APPLICATIONSETTINGEnum for APPLICATION_SETTING
        /// </summary>
        APPLICATIONSETTINGEnum = 13,
        
        /// <summary>
        /// Enum APPLICATIONCERTEnum for APPLICATION_CERT
        /// </summary>
        APPLICATIONCERTEnum = 14,
        
        /// <summary>
        /// Enum APPLICATIONPLACEMENTEnum for APPLICATION_PLACEMENT
        /// </summary>
        APPLICATIONPLACEMENTEnum = 15,
        
        /// <summary>
        /// Enum ACCOUNTEnum for ACCOUNT
        /// </summary>
        ACCOUNTEnum = 16,
        
        /// <summary>
        /// Enum ACCOUNTSETTINGEnum for ACCOUNT_SETTING
        /// </summary>
        ACCOUNTSETTINGEnum = 17,
        
        /// <summary>
        /// Enum GAMELEVELEnum for GAME_LEVEL
        /// </summary>
        GAMELEVELEnum = 18,
        
        /// <summary>
        /// Enum PACKEnum for PACK
        /// </summary>
        PACKEnum = 19,
        
        /// <summary>
        /// Enum MISSIONEnum for MISSION
        /// </summary>
        MISSIONEnum = 20,
        
        /// <summary>
        /// Enum TOURNAMENTEnum for TOURNAMENT
        /// </summary>
        TOURNAMENTEnum = 21,
        
        /// <summary>
        /// Enum ASSETEnum for ASSET
        /// </summary>
        ASSETEnum = 22,
        
        /// <summary>
        /// Enum ALBUMCONTESTEnum for ALBUM_CONTEST
        /// </summary>
        ALBUMCONTESTEnum = 23,
        
        /// <summary>
        /// Enum THEMEDESCRIPTOREnum for THEME_DESCRIPTOR
        /// </summary>
        THEMEDESCRIPTOREnum = 24,
        
        /// <summary>
        /// Enum OFFEREnum for OFFER
        /// </summary>
        OFFEREnum = 25,
        
        /// <summary>
        /// Enum OFFERLOCATIONEnum for OFFER_LOCATION
        /// </summary>
        OFFERLOCATIONEnum = 26,
        
        /// <summary>
        /// Enum EVENTEnum for EVENT
        /// </summary>
        EVENTEnum = 27,
        
        /// <summary>
        /// Enum RETAILEREnum for RETAILER
        /// </summary>
        RETAILEREnum = 28,
        
        /// <summary>
        /// Enum RETAILERLOCATIONEnum for RETAILER_LOCATION
        /// </summary>
        RETAILERLOCATIONEnum = 29,
        
        /// <summary>
        /// Enum NOTEEnum for NOTE
        /// </summary>
        NOTEEnum = 30,
        
        /// <summary>
        /// Enum CREATIVEEnum for CREATIVE
        /// </summary>
        CREATIVEEnum = 31,
        
        /// <summary>
        /// Enum FAVORITEEnum for FAVORITE
        /// </summary>
        FAVORITEEnum = 32,
        
        /// <summary>
        /// Enum LIKEEnum for LIKE
        /// </summary>
        LIKEEnum = 33,
        
        /// <summary>
        /// Enum RATINGEnum for RATING
        /// </summary>
        RATINGEnum = 34,
        
        /// <summary>
        /// Enum ANALYTICEnum for ANALYTIC
        /// </summary>
        ANALYTICEnum = 35,
        
        /// <summary>
        /// Enum THIRDPARTYCREDENTIALEnum for THIRD_PARTY_CREDENTIAL
        /// </summary>
        THIRDPARTYCREDENTIALEnum = 36,
        
        /// <summary>
        /// Enum THIRDPARTYNETWORKEnum for THIRD_PARTY_NETWORK
        /// </summary>
        THIRDPARTYNETWORKEnum = 37,
        
        /// <summary>
        /// Enum REGIONEnum for REGION
        /// </summary>
        REGIONEnum = 38
    }

    public RatableTypeEnum RatableType { get; set; }
    public long RatableId { get; set; }
    public CategoryResponse Category { get; set; }
    public Object Ratable { get; set; }
    public string SecondaryType { get; set; }
}


